scalardl-web-client-sdk.js

```diff
@@ -49,6 +53,57 @@ class ClientService extends ClientServiceBase {
 
     super(services, protobuf, properties);
   }
+
+  /**
+   * @description use indexedDB
+   *  and check if it is necessary to load keys for the users
+   */
+  async useIndexedDB() {
```

Comment:
Ideally this function is called in constructor eagerly or in each function lazily.
As you said, there is no way to have async constructor, but there seems some workaround for it.

https://gist.github.com/goloroden/c976971e5f42c859f64be3ad7fc6f4ed
https://qwtel.com/posts/software/async-constructor-pattern/

I'm sure you have thought about it but I want to explore a bit more for more user friendly API.

Reply:
Understood.
```
```